### DISCUSSION TITLE
Can I use Inline Conditional to hide/show columns?

### DISCUSSION THREAD

**Discussion by mathieu-lessard** [adobe/react-spectrum#4417] [Q&A] [answered]
I have some columns I only want to show sometimes, but I get the error "Cannot read properties of undefined (reading 'isRowHeader')" whenever I try to do it using an inline conditional.

See below link, click "Hide Vowels".
https://codesandbox.io/s/react-spectrum-toggled-columns-3dh589?file=/src/App.tsx

I can render an empty column: {!hideVowels ? (<Column key="i">I</Column>) : (<Column></Column>)}
Or I can move all my column data to an object array and remove that element from the array.
I can use hideHeader, but the column values still appear.

Is there a way to easily toggle columns without rewriting everything or rendering an empty column?

Thanks!

**Comment by snowystinger** [comment, marked as answer]
This is currently not supported but should be in the future. For now the way to accomplish this is to move to an object array and remove the columns you want when you don't want them to render.

**Reply by mathieu-lessard** [reply]
Thanks!
